src/dmd/backend/blockopt.d

```diff
@@ -2213,113 +2213,119 @@ private void blassertsplit()
         list_t bel = list_reverse(bl_enlist(b.Belem));
     L1:
         int dctor = 0;
-        foreach (el; ListRange(bel))
-        {
-            elem *e = list_elem(el);
 
-            int accumDctor(elem *e)
+        int accumDctor(elem *e)
+        {
+            while (1)
             {
-                while (1)
+                if (OTunary(e.Eoper))
                 {
-                    if (OTunary(e.Eoper))
-                    {
-                        e = e.EV.E1;
-                        continue;
-                    }
-                    else if (OTbinary(e.Eoper))
-                    {
-                        accumDctor(e.EV.E1);
-                        e = e.EV.E2;
-                        continue;
-                    }
-                    else if (e.Eoper == OPdctor)
-                        ++dctor;
-                    else if (e.Eoper == OPddtor)
-                        --dctor;
-                    break;
+                    e = e.EV.E1;
+                    continue;
                 }
-                return dctor;
+                else if (OTbinary(e.Eoper))
+                {
+                    accumDctor(e.EV.E1);
+                    e = e.EV.E2;
+                    continue;
+                }
+                else if (e.Eoper == OPdctor)
+                ++dctor;
+                else if (e.Eoper == OPddtor)
+                --dctor;
+                break;
             }
+            return dctor;
+        }
 
-            if (dctor == 0 &&   // don't split block between a dctor..ddtor pair
-                e.Eoper == OPoror && e.EV.E2.Eoper == OPcall && e.EV.E2.EV.E1.Eoper == OPvar)
-            {
-                Symbol *f = e.EV.E2.EV.E1.EV.Vsym;
-                if (f.Sflags & SFLexit)
-                {
-                    if (accumDctor(e.EV.E1))
-                    {
-                        accumDctor(e.EV.E2);
-                        continue;
-                    }
+        foreach (el; ListRange(bel))
+        {
+            elem *e = list_elem(el);
 
-                    // Create exit block
-                    ++numblks;
-                    maxblks += 3;
-                    block *bexit = block_calloc();
-                    bexit.BC = BCexit;
-                    bexit.Belem = e.EV.E2;
+            if (!(dctor == 0 &&   // don't split block between a dctor..ddtor pair
+                e.Eoper == OPoror && e.EV.E2.Eoper == OPcall && e.EV.E2.EV.E1.Eoper == OPvar))
+            {
+                accumDctor(e);
+                continue;
+            }
+            Symbol *f = e.EV.E2.EV.E1.EV.Vsym;
+            
```

Comment:
Trailing whitespace here:

```
Error - file 'dmd\backend\blockopt.d' contains trailing whitespace at line 2252
```

Maybe a good idea to take those 10 minutes to setup your editor to be aware of EditorConfig files.